solr/core/src/java/org/apache/solr/core/SolrCore.java

```diff
@@ -805,6 +792,24 @@ public SolrCore reload(ConfigSet coreConfig) throws IOException {
     }
   }
 
+  /**
+   * Clones the current core for core reload, with the provided CoreDescriptor and ConfigSet.
+   *
+   * @return the cloned core to be used for SolrCore#load
+   */
+  protected SolrCore cloneAsReloadCore(
```

Comment:
the name of this method reads weird to me... maybe cloneForReloadedCore sounds better